Stripe.hs

```diff
@@ -41,6 +42,19 @@ cancelSubscription (Just stripeId) = do
     forM_ subs $ \sub ->
         void $ S.cancelSubscription stripeId (S.subscriptionId sub) False
 
-findOrCreateCustomer :: S.TokenId -> Maybe S.CustomerId -> S.Stripe S.CustomerId
-findOrCreateCustomer _ (Just stripeId) = return stripeId
-findOrCreateCustomer token _ = S.customerId <$> S.createCustomerByToken token
+findOrCreateCustomer :: S.TokenId -> S.Email -> Maybe S.CustomerId -> S.Stripe S.CustomerId
+findOrCreateCustomer _ _ (Just stripeId) = return stripeId
+findOrCreateCustomer token email _ = S.customerId <$> S.createCustomerBase
```

Comment:
This sure looks like a good case for `def` and record syntax! (I know this isn't your code, though)